tendrl/commons/objects/index/__init__.py

```diff
@@ -0,0 +1,58 @@
+import json
+
+import etcd
+
+from tendrl.commons import objects
+from tendrl.commons.utils import etcd_utils
+
+
+class Index(objects.BaseObject):
+
+    def __init__(self, key=None, index_value=None,
+                 *args, **kwargs):
+        super(Index, self).__init__(*args, **kwargs)
+        self._index_name = key.strip("/").rsplit("/", 1)[-1]
+        self.value = "indexes/{0}".format(key.strip("/").rsplit("/", 1)[0])
+        self._is_stale = True
+        self._index_value = index_value
+
+        if self._index_value is None:
+            # index_value should be passed as None for tag based indexes
+            # other indexes can pass index_value.
+            curr_nodes = []
+            try:
+                curr_nodes = etcd_utils.read(key).value
+            except etcd.EtcdKeyNotFound:
+                pass
+
+            try:
+                curr_nodes = json.loads(curr_nodes)
+            except (ValueError, TypeError):
+                # No existing nodes have this tag index
+                pass
+
+            if NS.node_context.node_id in curr_nodes:
+                self._is_stale = False
+            else:
+                if "provisioner" in key:
```

Comment:
I feel,  every time if we read provisioner then provisioner node will change

Reply:
This is not an issue any way we have a mechanism to change provisioner node

Reply:
only one major doubt i have here looks like all attributes are private except value, we are not allowing to store private arrtibutes.

Reply:
@GowthamShanmugam but changing provisioner every time there is a read to find out who provisioner would cause issues as we have seen that in history.